observer/baseNodeProvider.go

```diff
@@ -28,6 +29,19 @@ func (bnp *baseNodeProvider) initNodes(nodes []*data.NodeData) error {
 	for _, observer := range nodes {
 		shardId := observer.ShardId
 		newNodes[shardId] = append(newNodes[shardId], observer)
+		isMeta := shardId == core.MetachainShardId
+		if isMeta {
+			continue
+		}
+
+		if shardId > bnp.numOfShards {
```

Comment:
greater or equal? 
3 shards -> no need for a node with shardId = 3

Reply:
right, updated

Reply:
ok